### PR TITLE
[tests-only][full-ci]Search files by content

### PR DESCRIPTION
**Pull request by nabim777**
## Description
This PR adds the API tests for full search api. The scenarios added in this PR are
- search file using context

## Related Issue
- https://github.com/owncloud/ocis/issues/6606#event-9635806756

## Motivation and Context
- there was no test coverage for api test for search file using context. so, this PR covers the require scenario test case

## How Has This Been Tested?
- test environment:
- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiGraph/fullSearch.feature [hunk 1]
```diff
@@ -89,3 +89,20 @@ Feature: full text search
       | old              |
       | new              |
       | spaces           |
+
+
+  Scenario Outline: search files using context
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/fullSearch.feature:94]: context or content?
```suggestion
  Scenario Outline: search files using content
```

File: tests/acceptance/features/apiGraph/fullSearch.feature [hunk 2]
```diff
@@ -89,3 +89,20 @@ Feature: full text search
       | old              |
       | new              |
       | spaces           |
+
+
+  Scenario Outline: search files using content
```
Comment (SwikritiT) [inline tests/acceptance/features/apiGraph/fullSearch.feature:94]:
```suggestion
  Scenario Outline: search files by content
```

File: tests/acceptance/config/behat.yml [hunk 3]
```diff
@@ -174,6 +174,21 @@ default:
         - OcisConfigContext:
         - PublicWebDavContext:
 
+    apiFullTextSearch:
+      paths:
+        - '%paths.base%/../features/apiFullTextSearch'
```
Comment (saw-jan) [inline tests/acceptance/config/behat.yml:179]: create a new `apiSearch` suite and bring other search feature files into this suite as well.
```suggestion
    apiSearch:
      paths:
        - '%paths.base%/../features/apiSearch'
```

Comment (SagarGi) [reply tests/acceptance/config/behat.yml:179]: or we have already `fullSearch.feature` we can simply add this to that feature. No need to create new suite.

Comment (grgprarup) [reply tests/acceptance/config/behat.yml:179]: > create a new `apiSearch` suite and bring other search feature files into this suite as well.

@saw-jan IMHO, This suite is added for full-text search which needs Tika service. Other search scenarios don't need Tika service so it does not worth running Tika service for all search scenarios. So suggested to create a separate suite.

Comment (grgprarup) [reply tests/acceptance/config/behat.yml:179]: > or we have already `fullSearch.feature` we can simply add this to that feature. No need to create new suite.

@SagarGi `fullSearch.feature` contains scenarios related to the tag. Searching through tag won't need Tika service. The refactoring of the file name will be done in the next PR.

File: .drone.star [hunk 4]
```diff
@@ -755,7 +763,8 @@ def localApiTestPipeline(ctx):
                             },
                             "steps": skipIfUnchanged(ctx, "acceptance-tests") +
                                      restoreBuildArtifactCache(ctx, "ocis-binary-amd64", "ocis/bin") +
-                                     ocisServer(storage, params["accounts_hash_difficulty"], extra_server_environment = params["extraServerEnvironment"], with_wrapper = True) +
+                                     (tikaService() if params["tikaNeeded"] else []) +
+                                     (ocisServer(storage, params["accounts_hash_difficulty"], extra_server_environment = params["extraServerEnvironment"], with_wrapper = True, tika_enabled = True) if params["tikaNeeded"] else ocisServer(storage, params["accounts_hash_difficulty"], extra_server_environment = params["extraServerEnvironment"], with_wrapper = True)) +
```
Comment (SwikritiT) [inline .drone.star:767]: Will this achieve what you're trying to get? 
```suggestion
                                     ocisServer(storage, params["accounts_hash_difficulty"], extra_server_environment = params["extraServerEnvironment"], with_wrapper = True, tika_enabled = params["tikaNeeded"]
```

File: tests/acceptance/config/behat.yml (hunk 3 above)
Comment (saw-jan) [reply tests/acceptance/config/behat.yml:180]: Ahh! That's right. At the time of reviewing this, there was no tika mentioned.
:+1:

File: tests/acceptance/features/apiFullTextSearch/search.feature [hunk 5]
```diff
@@ -0,0 +1,26 @@
+@api
+Feature: full text search
+  As a user
+  I want to do full text search
+  So that I can find the files with the content I am looking for
+
+  Background:
+    Given user "Alice" has been created with default attributes and without skeleton files
+
+  Scenario Outline: search files by content
```
Comment (saw-jan) [inline tests/acceptance/features/apiFullTextSearch/search.feature:10]: new line for tags
```suggestion


  Scenario Outline: search files by content
```

File: .drone.star (hunk 4 above)
Comment (nabim777) [reply .drone.star:767]: Thanks!! It is pretty beneficial.

**Review by saw-jan — APPROVED** [review]
LGTM

File: tests/acceptance/features/apiFullTextSearch/search.feature [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    And user "Alice" has uploaded file with content "hello world from nepal" to "file1.txt"
+    And user "Alice" has uploaded file with content "saying hello to the world" to "file2.txt"
+    And user "Alice" has uploaded file with content "nepal want to say hello" to "file3.txt"
+    And user "Alice" has uploaded file with content "namaste from nepal" to "hello.txt"
+    When user "Alice" searches for "Content:hello" using the WebDAV API
+    Then the HTTP status code should be "207"
+    And the search result of user "Alice" should contain only these files:
+      | file1.txt |
+      | file2.txt |
+      | file3.txt |
```
Comment (grgprarup) [inline tests/acceptance/features/apiFullTextSearch/search.feature:22]:
```suggestion
      | keywordAtStart.txt  |
      | keywordInMiddle.txt |
```

File: tests/acceptance/features/apiFullTextSearch/search.feature [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  Background:
+    Given user "Alice" has been created with default attributes and without skeleton files
+
+
+  Scenario Outline: search files by content
+    Given using <dav-path-version> DAV path
+    And user "Alice" has uploaded file with content "hello world from nepal" to "file1.txt"
+    And user "Alice" has uploaded file with content "saying hello to the world" to "file2.txt"
+    And user "Alice" has uploaded file with content "nepal want to say hello" to "file3.txt"
+    And user "Alice" has uploaded file with content "namaste from nepal" to "hello.txt"
```
Comment (grgprarup) [inline tests/acceptance/features/apiFullTextSearch/search.feature:16]:
```suggestion
    And user "Alice" has uploaded file with content "hello keyword at start" to "keywordAtStart.txt"
    And user "Alice" has uploaded file with content "keyword hello in middle" to "keywordInMiddle.txt"
    And user "Alice" has uploaded file with content "keyword in the file name" to "hello.txt"
```
IMO, it clarifies the purpose of creating different files more effectively.

File: tests/acceptance/features/apiFullTextSearch/search.feature (hunk 6 above)
Comment (nabim777) [reply tests/acceptance/features/apiFullTextSearch/search.feature]: ok 👌

**Review by grgprarup — COMMENTED** [review]
lgtm :+1:

**Pull request merged**